lib/phoenix_ecto/check_repo_status.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Phoenix.Ecto.CheckRepoStatus do
+  @moduledoc """
+  A plug that does some checks on your application repos.
+
+  Checks if the storage is up (database is created) or if there are any pending migrations.
+  Both checks can raise an error if the conditions are not met.
+
+  ## Plug options
+
+    * `:otp_app` - name of the application which the repos are fetched from
+    * `:get_migration_function` - function that returns the migrations for a given `repo`.
+      Should return a list of tuples with three elements,
+      e.g. `[{state :: :up | :down, version :: integer(), name :: String.t()]]`
+      If `ecto_sql` dependency is loaded, uses `Ecto.Migrator.migrations/1` by default.
+  """
+
+  @behaviour Plug
+
+  alias Plug.Conn
+
+  def init(opts), do: opts
```

Comment:
Maybe we should check if `:otp_app` key was given on `init` and raise otherwise.

Reply:
[fixed here](https://github.com/phoenixframework/phoenix_ecto/compare/d85122ccded004efe2b01f32d66e64073c16766f..6e01ed7b851cbbfd63fa25a207832a82254452f2)